salt/modules/network.py

```diff
@@ -1322,10 +1323,17 @@ def ip_addrs6(interface=None, include_loopback=False, cidr=None):
     addrs = __utils__["network.ip_addrs6"](
         interface=interface, include_loopback=include_loopback
     )
+
     if cidr:
         return [i for i in addrs if __utils__["network.in_subnet"](cidr, [i])]
-    else:
-        return addrs
+
+    if type == "public":
```

Comment:
alternately, we can define the lists of addresses inside of each if statement and then have a single return at the bottom (for a function of this length/complexity, i'm not sure it matters, though)